examples/server/server.cpp

```diff
@@ -2591,11 +2592,22 @@ int main(int argc, char ** argv) {
         return false;
     };
 
-    auto middleware_server_state = [&res_error, &state](const httplib::Request &, httplib::Response & res) {
+    auto middleware_server_state = [&res_error, &state](const httplib::Request & req, httplib::Response & res) {
         server_state current_state = state.load();
         if (current_state == SERVER_STATE_LOADING_MODEL) {
-            res_error(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
-            return false;
+            httplib::Request & modified_req = (httplib::Request &) req;
+            const char* path_c = modified_req.path.c_str();
+            int path_c_len = strlen(path_c);
+            char last_five[6];
+            strcpy(last_five, path_c + (path_c_len -5));
+
+            if ((strcmp(path_c, "/") == 0) || (strcmp(last_five, ".html") == 0)) {
+                modified_req.path = "/loading.html";
```

Comment:
IMO modifying path is not a very clean solution (even when we're inside `set_pre_routing_handler`). You can simply call `res.set_content`

Reply:
another solution could be to move `handle_static_file` before middleware, then use it inside middleware

Reply:
Yeah I agree. As you suggested, I simplified the diff and used res.set_content() instead, which I think is a much cleaner solution.